### PR TITLE
wrap: submodule fixes

### PR DESCRIPTION
**Pull request by ernestask**
These changes seem to work well enough.

### CODE REVIEW

File: mesonbuild/wrap/wrap.py [hunk 1]
```diff
@@ -150,16 +151,16 @@ def resolve(self, packagename):
...
             return False
         # Is `dirname` a submodule?
-        ret, out = quiet_git(['submodule', 'status', dirname])
+        ret, out = quiet_git(['submodule', 'status', dirname], self.subdir_root)
         if not ret:
             return False
         # Submodule has not been added, add it
         if out.startswith(b'-'):
-            if subprocess.call(['git', 'submodule', 'update', dirname]) != 0:
+            if subprocess.call(['git', '-C', self.subdir_root, 'submodule', 'update', '--init', dirname]) != 0:
```
Comment (nirbheek) [inline mesonbuild/wrap/wrap.py]: Does `--init` work even with submodules that have been initialized already but were removed later (manually)?

Comment (ernestask) [inline mesonbuild/wrap/wrap.py]: As far as I understand (and have experienced), it does. It’s not much different from having `git submodule init` as a separate step, and it by itself doesn’t do a whole lot after the initial registration.

**Comment by jpakkane** [comment]
This message is posted to every outstanding merge request.

We have transitioned from Github wiki to our new documentation system that generates all documentation directly from the Git repo. This means that from now on every merge request must come with corresponding documentation changes.

If your MR requires documentation changes, do the necessary changes to files in `docs/markdown`.

If your MR adds a new feature, add a description to the release notes in `docs/markdown/Release-notes-for-0.41.0.md`.

If this commit is just a bugfix with no functional changes, you don't need to do anything.

Thank you

**Comment by jpakkane** [comment]
Please add yourself at the end of `authors.txt` and we'll get this merged.

**Pull request merged**